compiler/hash-lexer/src/lib.rs

```diff
@@ -537,28 +530,25 @@ impl<'a> Lexer<'a> {
                             Span::new(start, self.offset.get()),
                         ),
                         Ok(parsed) => {
-                            let (value, sign) =
-                                if negated { (-parsed, Sign::Minus) } else { (parsed, Sign::None) };
-
                             // Create interned float constant
-                            let float_const = CONSTANT_MAP.create_float_constant(value, suffix);
-                            TokenKind::FloatLit(sign, float_const)
+                            let float_const = if let Some(suffix_ident) = suffix && suffix_ident == CORE_IDENTIFIERS.f32 {
+                                CONSTANT_MAP.create_f32_float_constant(parsed as f32, suffix)
+                            } else {
+                                CONSTANT_MAP.create_f64_float_constant(parsed, suffix)
```

Comment:
Don't you need to check that the suffix is either empty or f64?

Reply:
invalid suffixes are later caught by the parser, and if there is no suffix then it is automatically assumed as `f64`.